File: Java/Square.java
public class Square extends Rectangle implements Equilateral {
    public Square() {
        this(1.0);
    }
    public Square(double sideLength) {
        super(sideLength, sideLength);
    }
    public double getSideLength() {
        assert width == height;
        return width;
    }

    public static void main(String[] args) {
        Square sq = new Square();
        System.out.println("The perimeter of the unit square is " + sq.getPerimeter());
        System.out.printf("Squares have %d diagonals.\n", sq.getNumDiagonals());
        System.out.println("The circle that circumscribes the unit square has a circumference of "+
                           sq.circumscribe().getCircumference());
    }
}

File: C++/square.cc
#include "square.h"
#include "circle.h"

Square::Square() : Rectangle(1, 1) {
}

Square::Square(float size) : Rectangle(size, size) {
}

float Square::getSideLength() const {
    return width_;
}

void Square::setSideLength(float length) {
    width_ = height_ = length;
}

int Square::getNumSides() const {
    return Rectangle::getNumSides();
}

float Square::getPerimeter() const {
    return Equilateral::getPerimeter();
}

int Square::getNumDiagonals() const {
    return Equilateral::getNumDiagonals();
}

void Square::inscribe(Circle& round) {
    setSideLength(round.getRadius() * 2);
}